csharp/IceDiscovery/Greeter/Client/Program.cs

```diff
@@ -3,16 +3,13 @@
 // Slice module VisitorCenter in Greeter.ice maps to C# namespace VisitorCenter.
 using VisitorCenter;
 
-// Configure the communicator to load the IceDiscovery plugin during initialization. This plugin installs a default
-// locator on the communicator.
-var initData = new Ice.InitializationData
-{
-    properties = new Ice.Properties(ref args)
-};
-initData.properties.setProperty("Ice.Plugin.Discovery", "IceDiscovery:IceDiscovery.PluginFactory");
+// Register the IceDiscovery plugin. The plugin will be loaded during communicator initialization.
+IceDiscovery.Util.registerIceDiscovery(loadOnInitialize: true);
```

Comment:
We should fix the C# code to make loadOnInitialize = true by default like in C++.